### PR TITLE
fix(files_versions): renaming file version when its not a string

### LINKED ISSUE #45642: [Bug]: Naming File Version a number removes ability to edit version
### ⚠️ This issue respects the following points: ⚠️

- [X] This is a **bug**, not a question or a configuration/webserver/proxy issue.
- [X] This issue is **not** already reported on [Github](https://github.com/nextcloud/server/issues?q=is%3Aopen+is%3Aissue+label%3Abug) OR [Nextcloud Community Forum](https://help.nextcloud.com/) _(I've searched it)_.
- [X] Nextcloud Server **is** up to date. See [Maintenance and Release Schedule](https://github.com/nextcloud/server/wiki/Maintenance-and-Release-Schedule) for supported versions.
- [X] I agree to follow Nextcloud's [Code of Conduct](https://nextcloud.com/contribute/code-of-conduct/).

### Bug description

When naming the version of a file to a number: e.g. 1.0
1) it gets renamed to 1 after refresh
2) the version can't be renamed as the input field is not rendered

Console throws:
t.innerversionlabel.trim is not a function
and: 
this.$refs.labelInput is not defined

### Steps to reproduce

1. create a file
2. open sharing side menu
3. go to versions
4. click `edit version name` in three dots menu
5. insert a float
6. reload page
7. try to edit version name

### Expected behavior

version name should stay editable
probably 1.0 should not be renamed to 1

### Nextcloud Server version

27, 28,29 master

### PR DESCRIPTION
**Pull request by sanskar-soni-9**
* Resolves: #45642 

## Summary
Naming a file version to a number (e.g. 1 or 1.0) is received as number type in the [response object](https://github.com/nextcloud/server/blob/1a6440c353d3c797140215e4ab885544f1ac3741/apps/files_versions/src/utils/versions.ts#L42-L45) breaking the renaming of version as it needed to be a string

## Todo
- [ ] Get the correct version name when its in float ending with '0' (e.g. 2.0 becomes 2 but 2.1 - 2.9 it will show correct version)

#### Before
240609_09h27m35s_screenshot
#### After
240609_09h28m45s_screenshot

## Checklist

- Code is [properly formatted](https://docs.nextcloud.com/server/latest/developer_manual/digging_deeper/continuous_integration.html#linting)
- [Sign-off message](https://github.com/src-d/guide/blob/master/developer-community/fix-DCO.md) is added to all commits
- [ ] Tests ([unit](https://docs.nextcloud.com/server/latest/developer_manual/app_development/tutorial.html#unit-tests), [integration](https://docs.nextcloud.com/server/latest/developer_manual/app_development/tutorial.html#integration-tests), api and/or acceptance) are included
- [x] Screenshots before/after for front-end changes
- [x] Documentation ([manuals](https://github.com/nextcloud/documentation/) or wiki) has been updated or is not required
- [x] [Backports requested](https://github.com/nextcloud/backportbot/#usage) where applicable (ex: critical bugfixes)

### CODE REVIEW

**Review by artonge — APPROVED** [review]

File: apps/files_versions/src/utils/versions.ts [hunk 1]
```diff
@@ -91,7 +91,7 @@ function formatVersion(version: any, fileInfo: any): Version {
 
 	return {
 		fileId: fileInfo.id,
-		label: version.props['version-label'],
+		label: version.props['version-label'].toString(),
```
Comment (susnux) [inline apps/files_versions/src/utils/versions.ts:94]: I think using the string constructor is a bit better understandable and prevents `TypeError` on some cases.

```suggestion
		label: String(version.props['version-label']),
```

Comment (sanskar-soni-9) [reply apps/files_versions/src/utils/versions.ts]: fixed!, thanks :)

**Review by susnux — APPROVED** [review]

**Comment by sanskar-soni-9** [comment]
Its failing, do I need to fix anything?

**Comment by artonge** [comment]
> Its failing, do I need to fix anything?

Can you compile and commit the build files?

**Comment by sanskar-soni-9** [comment]
> > Its failing, do I need to fix anything?
> 
> Can you compile and commit the build files?

is this correct?

**Comment by artonge** [comment]
@sanskar-soni-9 and now the DCO :)

**Comment by sanskar-soni-9** [comment]
> @sanskar-soni-9 and now the DCO :) Also, CI is red, but probably unrelated to your changes, it should be fixed soon :).

sorry forgot to sign commit, in CI I saw its saying to run composer update should I do that and add those files too?

**Comment by artonge** [comment]
> sorry forgot to sign commit, in CI I saw its saying to run composer update should I do that and add those files too?

No, this is the error I am referring to in my last comment. Ignore it for now, and we'll try to fix it on our side :).

**Comment by sanskar-soni-9** [comment]
> > sorry forgot to sign commit, in CI I saw its saying to run composer update should I do that and add those files too?
> 
> No, this is the error I am referring to in my last comment. Ignore it for now, and we'll try to fix it on our side :).

Okay, btw do I need to add build files with each PR. And Thanks a lot for helping 😊

**Comment by sanskar-soni-9** [comment]
> In most repo, yes :). In doubt, just check the CI.

Thanks for clarifying, will keep that in mind from next time :)

**Comment by artonge** [comment]
@sanskar-soni-9 can you rebase? You'll need to `npm ci && npm run build` during the conflict resolution.

**Comment by sanskar-soni-9** [comment]
@artonge done :)

**Comment by susnux** [comment]
@sanskar-soni-9 we need to merge this into a branch first to run the CI, but we will merge into master when green.
Thank you so much :)

**Comment by sanskar-soni-9** [comment]
undefined case fixed
240619_21h40m56s_screenshot

**Pull request merged**